sdk/containerregistry/azure-containerregistry/azure/containerregistry/_container_registry_client.py

```diff
@@ -885,13 +873,12 @@ def download_manifest(self, repository, tag_or_digest, **kwargs):
         return DownloadManifestResult(digest=digest, data=manifest_stream, manifest=manifest)
 
     @distributed_trace
-    def download_blob(self, repository, digest, **kwargs):
-        # type: (str, str, **Any) -> DownloadBlobResult | None
+    def download_blob(self, repository: str, digest: str, **kwargs) -> Optional[DownloadBlobResult]:
         """Download a blob that is part of an artifact.
 
         :param str repository: Name of the repository
         :param str digest: The digest of the blob to download.
-        :returns: DownloadBlobResult or None
+        :returns: DownloadBlobResult or None when the download result is empty.
```

Comment:
Well, after checking the implementation of `get_blob()` operation, we don't have any chance to get an empty result, it will raise an error when the response status code isn't _200_ or _307_ inside the call. But **mypy** still think the return could be _None_ as `deserialized` is initialized with None in get_blob(), we need to ignore this error here.

Reply:
Update:
When return code is `307`, will return a redirect address to get the data. Here's an example: https://github.com/Azure/azure-rest-api-specs/blob/main/specification/containerregistry/data-plane/Azure.ContainerRegistry/stable/2021-07-01/examples/GetBlob.json#L14 
I haven't met 307 returned so far, it probably redirect automatically by the `default redirect policy`(_azure.core.pipeline.policies.RedirectPolicy_). 
The return value should not be None.